python/lance/data/convert/nuscenes.py

```diff
@@ -0,0 +1,267 @@
+            pa.field("token", pa.string()),
+            pa.field("timestamp", pa.int64()),
+            pa.field("log_token", pa.string()),
+            pa.field("key_camera_token", pa.string()),
+            
+            # Log
+            pa.field("log_token_", pa.string()),
+            pa.field("log_logfile_", pa.string()),
+            pa.field("log_vehicle_", pa.string()),
+            pa.field("log_date_captured_", pa.string()),
+            pa.field("log_location_", pa.string()),
+            
+            # Calibrated Sensor
+            pa.field("calibrated_sensor_token_", pa.string()),
+            pa.field("calibrated_sensor_sensor_token_", pa.string()),
+            pa.field("calibrated_sensor_translation_", pa.list_(pa.float32(), 3)),
+            pa.field("calibrated_sensor_rotation_", pa.list_(pa.float32(), 4)),
+            pa.field("calibrated_sensor_camera_intrinsic_", pa.list_(pa.list_(pa.float32(), 3))),
+            pa.field("calibrated_sensor_camera_distortion_", pa.list_(pa.float32())), # can be 5 or 6 length
+            
+            # Sensor
+            pa.field("sensor_token_", pa.string()),
+            pa.field("sensor_channel_", pa.string()),
+            pa.field("sensor_modality_", pa.string()),
+
+            # Sample Data
+            pa.field("sample_data_token_", pa.string()),
+            pa.field("sample_data_sample_token_", pa.string()),
+            pa.field("sample_data_ego_pose_token_", pa.string()),
+            pa.field("sample_data_calibrated_sensor_token_", pa.string()),
+            pa.field("sample_data_filename_", pa.string()),
+            pa.field("sample_data_fileformat_", pa.string()),
+            pa.field("sample_data_width_", pa.int32()),
+            pa.field("sample_data_height_", pa.int32()),
+            pa.field("sample_data_timestamp_", pa.int64()),
+            pa.field("sample_data_is_key_frame_", pa.bool_()),
+            pa.field("sample_data_next_", pa.string()),
+            pa.field("sample_data_prev_", pa.string()),
+
+            # Ego Pose
+            pa.field("ego_pose_token_", pa.string()),
+            pa.field("ego_pose_translation_", pa.list_(pa.float32(), 3)),
+            pa.field("ego_pose_rotation_", pa.list_(pa.float32(), 4)),
+            pa.field("ego_pose_timestamp_", pa.int64()),
+            pa.field("ego_pose_rotation_rate_", pa.list_(pa.float32(), 3)),
+            pa.field("ego_pose_acceleration_", pa.list_(pa.float32(), 3)),
+            pa.field("ego_pose_speed_", pa.float32()),
+
+            # Annotations
+            pa.field("surface_ann", pa.list_(surface_ann_schema)),
+            pa.field("object_ann", pa.list_(object_ann_schema))
+        ])
+    
+    def image_uris(self, table) -> List[str]:
+        """Return image uris to read the binary column"""
+        uris = [os.path.join(self.uri_root, image_uri)
+            for image_uri in table["sample_data_filename_"].to_numpy()]
```

Comment:
why do we need `to_numpy()` to iterate over filenames? what is the original type of `table["sample_data_filtername_"]`?

Reply:
os.path.join requires str, or a blob (bytes)
the original type of `sample_data_filename` is a `StringScalar`